### PR TITLE
feat: RND-116: YOLOv8 ML Backend

### PR DESCRIPTION
**Pull request by makseq**
See README.md in label_studio_ml/examples/yolo folder.

### CODE REVIEW

File: label_studio_ml/examples/mmdetection-3/requirements-base.txt [hunk 1]
```diff
@@ -1,2 +1,2 @@
 gunicorn==22.0.0
-label-studio-ml @ git+https://github.com/HumanSignal/label-studio-ml-backend.git
+label-studio-ml @ git+https://github.com/HumanSignal/label-studio-ml-backend.git@fix/rnd-117
```
[the remark this answers is not in the document]
Comment (makseq) [reply label_studio_ml/examples/mmdetection-3/requirements-base.txt:2]: Yes, you absolutely right

File: label_studio_ml/examples/yolo/control_models/choices.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    def is_control_matched(cls, control) -> bool:
+        # check object tag type
+        if control.objects[0].tag != 'Image':
+            return False
+        # support both Choices and Taxonomy because of their similarity
+        return control.tag in [cls.type, 'Taxonomy']
+
+    def predict_regions(self, path) -> List[Dict]:
+        results = self.model.predict(path)
+        self.debug_plot(results[0].plot())
```
[the remark this answers is not in the document]
Comment (makseq) [reply label_studio_ml/examples/yolo/control_models/choices.py:28]: Yes, there is DEBUG_PLOT env car inside of this function

File: label_studio_ml/examples/yolo/README.md [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+
+This guide describes the simplest way to start using YOLO ML backend with Label Studio.
+
+# Quick Start
+
+1. Add `LABEL_STUDIO_URL` and `LABEL_STUDIO_API_KEY` to the `docker-compose.yml` file. 
+
+2. Run docker compose
+
+    ```
```
[the remark this answers is not in the document]
Comment (makseq) [reply label_studio_ml/examples/yolo/README.md:11]: yes, you need to build it on your side.

File: label_studio_ml/examples/yolo/README.md [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+```
+
+# Run YOLO ML backend
+
+## Running with Docker (Recommended)
+
+1. Start Machine Learning backend on `http://localhost:9090` with prebuilt image:
+
+```bash
+docker-compose up
```
[the remark this answers is not in the document]
Comment (makseq) [reply label_studio_ml/examples/yolo/README.md:318]: yeh, build is needed. I haven't yet inspected this part of readme thoroughly

File: label_studio_ml/examples/yolo/control_models/base.py [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+os.makedirs(MODEL_ROOT, exist_ok=True)
+# if true, allow to use custom model path from the control tag in the labeling config
+ALLOW_CUSTOM_MODEL_PATH = os.getenv('ALLOW_CUSTOM_MODEL_PATH', "false").lower() in ["1", "true"]
+
+# Global cache for YOLO models
+_model_cache = {}
+logger = logging.getLogger(__name__)
+
+
+class ControlModel(BaseModel):
```
[the remark this answers is not in the document]
Comment (makseq) [reply label_studio_ml/examples/yolo/control_models/base.py:27]: 1. We can't have more then one LabelStudioMLBase in model.py: 
https://github.com/HumanSignal/label-studio-ml-backend/blob/master/label_studio_ml/server.py#L119

2. My idea is to have only one ML backend that process all control tags from your labeling configuration. This way you can combine different control tags in one labeling configuration. E.g. 
```
<View>

  <Image name="image" value="$image" zoom="true"/>
  
  <RectangleLabels name="rect" toName="image" score_threshold="0.1">
    <Label value="plane" background="red" predicted_values="plane,helicopter"/>
  </RectangleLabels>
 
 <Choices name="choices" toName="image">
   <Choice value="car">
 </Choices>
  
</View>
```

Yolo will be used for RectangleLabels and Choices simultaneously. 

This is not achievable If you use two different LabelStudioMLBase instances.

File: label_studio_ml/examples/yolo/cli.py [hunk 6]
```diff
@@ -0,0 +1,142 @@
+import os
```
[the remark this answers is not in the document]
Comment (makseq) [reply label_studio_ml/examples/yolo/cli.py:1]: docs added!

**Review by caitlinwheeless — APPROVED** [review]

**Review by micaelakaplan — APPROVED** [review]

**Pull request merged**